Lib/test/test_re.py

```diff
@@ -700,6 +700,39 @@ def test_other_escapes(self):
             with self.subTest(c):
                 self.assertRaises(re.error, re.compile, '[\\%c]' % c)
 
+    def test_named_unicode_escapes(self):
+        # test individual Unicode named escapes
+        suites = [
+            [   # basic matches
+                ['\u2014', r'\u2014', '\N{EM DASH}',
+                 r'\N{EM DASH}'],                               # pattern
+                ['\u2014', '\N{EM DASH}', '—', '—and more'],    # matches
+                ['\u2015', '\N{EN DASH}']                       # no match
+            ],
+            [   # character set matches
+                ['[\u2014-\u2020]', r'[\u2014-\u2020]',
+                 '[\N{EM DASH}-\N{DAGGER}]', r'[\N{EM DASH}-\N{DAGGER}]',
+                 '[\u2014-\N{DAGGER}]', '[\N{EM DASH}-\u2020]',],                               # pattern
+                ['\u2014', '\N{EM DASH}', '—', '—and more', '\u2020',
+                 '\N{DAGGER}', '†', '\u2017', '\N{DOUBLE LOW LINE}'],
+                ['\u2011', '\N{EN DASH}', '\u2013', 'xyz', '\u2021']
+            ],
+        ]
+
+        for patterns, match_yes, match_no in suites:
+            for pat in patterns:
+                for target in match_yes:
+                    self.assertTrue(re.match(pat, target))
```

Comment:
Use `subTest()` when check in a loop.

Actually I think that loops are not needed. It is enough to test just one case for a pattern.